tools/tools/nanobsd/Files/root/updatep1

```diff
@@ -51,3 +52,14 @@ dd of=/dev/${NANO_DRIVE}s1 obs=64k
 fsck_ffs -n /dev/${NANO_DRIVE}s1a
 
 gpart set -a active -i 1 ${NANO_DRIVE}
+#
+# KD - If the partition has an EFI partition then set "loader.env"
+#
+EFI=`gpart show ${NANO_DRIVE}|grep 'efi'`
```

Comment:
since we're running on the system in question, you can look at `sysctl -n machdep.bootmethod` to know if we've booted with UEFI.

Reply:
True but I need the partition anyway and if its not on the drive for some reason (the /etc/nanobsd.conf file is wrong) it won't work, or if you use the media in two boxes, and the one you're on right now isn't EFI, it still should set it correctly.  That's actually relevant for me in that I have a higher-performance EFI-boot-only firewall but if it blows up my older pcEngines box will work (at reduced performance) and being able to yank the stick out of one and shove it in the other and have it come up and run is nice.